MdePkg/Library/PeiServicesLib/PeiServicesLib.c

```diff
@@ -685,7 +685,11 @@ InternalPeiServicesInstallFvInfoPpi (
   }
 
   FvInfoPpiDescriptor = AllocatePool (sizeof (EFI_PEI_PPI_DESCRIPTOR));
-  ASSERT (FvInfoPpiDescriptor != NULL);
+  if (FvInfoPpiDescriptor == NULL) {
+    ASSERT (FvInfoPpiDescriptor != NULL);

```

Comment:
What issue is resolved here? Does it mean each ASSERT usage should be updated like this?

Reply:
https://github.com/tianocore/edk2/blob/master/MdePkg/Include/Library/DebugLib.h#L395

ASSERTS can be disabled, which makes them a no-op.

In the case that asserts are disabled, then the system has the ability to proceed with a failed allocation and dereference the pointer.

This change ensures that if the allocation fails, and asserts are disabled, that the system will not access the invalid pointer that was returned.

Reply:
There are many similar usages in Edk2 code. Now, you only update MdePkg. Do you plan to update others, such as MdeModulePkg?

Reply:
This PR is just to address codeql issues that were reported in the MdePkg.

As other codeql patches are added for other packages, I believe they would address these in the same way.

It really just comes down to the code that was written;  If the code expects an ASSERT to prevent dereferencing a NULL pointer then it should be updated to prevent it. If the code was written to where a NULL pointer is an acceptable input, then it doesn't need the same update, as above.

Reply:
Will codeql report the code issue for ASSERT usage?  If the code doesn't need to be updated, how handle it?

Reply:
No, codeQL will only report an error for unguarded references to NULL variables. CodeQL doesn't look at assert as the stop of execution because asserts can be disabled based of pcd settings.

If codeQL can find a path through the source code that would result in a NULL bring dereferenced, it will report an issue for it. Sometimes it takes closer examination of the code to be able to find the path that would cause the dereference, but I has been there in everything I've examined.

Reply:
OK. I have no other comment here.